File: src/training/Chapter 1/.2 Ad Hoc/friday.py
"""
ID: aaronli2
LANG: PYTHON3
TASK: friday
"""
import sys
import pdb; pdb.set_trace()
sys.stdin=open("friday.in",'r')
sys.stdout=open("friday.out",'w')
def dim(month, year):
    if month == 1:
        return 28 if (year % 4) == 1 or ((year % 100) == 0 and ((year+300) % 400) == 1) else 29
    return {
        8: 30,
        3: 30,
        5: 30,
        10: 30
    }.get(month, 31)


n = eval(input())
if n < 0:
    n = 0
if n > 400:
    n = 400
res = [0, 0, 0, 0, 0, 0, 0]
dow = 0
for i in range(0, n):
    for m in range(0, 12):
        res[(dow+6) % 7] += 1
        dow = (dow+dim(m, i)) % 7
end2 = " "
for i in range(0, 7):
    if i == 6:
        end2 = "\n"
    print(res[(i+6) % 7], end=end2)

File: src/training/Chapter 1/.2 Ad Hoc/friday.cpp
/*
ID: aaronli2
TASK: friday
LANG: C++
*/
#include <iostream>
#include <fstream>
#include <string>

using namespace std;
int daysinmonth(int month, int year)
{
	if (month == 1)
		return ((year % 4) || (!(year % 100) && ((year + 300) % 400))) ? 28 : 29;
	switch (month)
	{
	case 8:
	case 3:
	case 5:
	case 10:
		return 30;
	default:
		return 31;
	}
}

int main()
{
	ofstream fout("friday.out");
	ifstream fin("friday.in");
	int n;
	fin >> n;
	if (n < 0)
		n = 0;
	if (n > 400)
		n = 400;
	int result[7] = {0};
	int dow = 0, month = 0;
	for (int i = 0; i < n; ++i)
		for (int m = 0; m < 12; ++m)
		{
			++result[((dow + 6) % 7)];
			dow = (dow + daysinmonth(m, i)) % 7;
		}
	for (int i = 0; i < 7; ++i)
	{
		fout << result[(i + 6) % 7];
		if (i == 6)
			fout << endl;
		else
			fout << " ";
	}
	return 0;
}
